lib/complex.ex

```diff
@@ -58,20 +67,51 @@ defmodule Complex do
 
   #### Examples
       iex> Complex.parse("1.1+2.2i")
-      %Complex{im: 2.2, re: 1.1}
+      {%Complex{im: 2.2, re: 1.1}, ""}
+
+      iex> Complex.parse("1+2i")
+      {%Complex{im: 2.0, re: 1.0}, ""}
+
+      iex> Complex.parse("2-3i")
+      {%Complex{im: -3.0, re: 2.0}, ""}
+
+      iex> Complex.parse("-1.0-3.i")
+      {%Complex{im: -3.0, re: -1.0}, ""}
+
+      iex> Complex.parse("-1.0+3.i 2.2+3.3i")
+      {%Complex{im: 3.0, re: -1.0}, " 2.2+3.3i"}
+
+      iex> Complex.parse("1e-4-3e-3i")
+      {%Complex{im: -3.0e-3, re: 1.0e-4}, ""}
   """
-  @spec parse(String.t()) :: complex
+  @spec parse(String.t()) :: {complex, String.t()} | :error
   def parse(str) do
-    [_, real, imag] = Regex.run(~r/([-]?\d+\.\d+)\s*\+\s*([-]?\d+\.\d+)i/, str)
+    case Regex.run(
+           ~r/([\-\+]?\d+(?:\.?\d*|e[\+\-]?\d+))([\-\+]\d+(?:\.?\d*|e[\+\-]?\d+))i(.*)/,
+           str,
+           capture: :all_but_first
+         ) do
```

Comment:
I would personally prefer to parse using Float.parse/Integer.parse plus some basic parsing around the operator, but that can be handled in a separate PR. :)

Reply:
Resolving in favor of #9